client/executor/src/native_executor.rs

```diff
@@ -306,15 +333,74 @@ impl RuntimeVersionOf for WasmExecutor {
 	}
 }
 
+/// A wrapper which merges two sets of host functions, and allows the second set to override
+/// the host functions from the first set.
+pub struct ExtendedHostFunctions<Base, Overlay> {
+	phantom: PhantomData<(Base, Overlay)>,
+}
+
+impl<Base, Overlay> HostFunctions for ExtendedHostFunctions<Base, Overlay>
+where
+	Base: HostFunctions,
+	Overlay: HostFunctions,
+{
+	fn host_functions() -> Vec<&'static dyn Function> {
+		let mut base = Base::host_functions();
+		let overlay = Overlay::host_functions();
+		base.retain(|host_fn| {
+			!overlay.iter().any(|ext_host_fn| host_fn.name() == ext_host_fn.name())
```

Comment:
do we want to pull the overlay names into a set or do we think there will be very few of them?

Reply:
That's how it was originally, so it's probably not a big deal? Nevertheless there aren't that many of them, and I haven't seen this pop up in any of profiles.

I can rework this to use a `HashSet` if you really want though.